### ISSUE TITLE
TO Golang: api.NewInfo Close() deferral postpones DB transaction commit until after API response is written

### ISSUE THREAD

**Issue by rawlinp** [apache/trafficcontrol#2666]
General usage of `api.NewInfo()` [1] is that the `APIInfo.Close()` method is immediately deferred in the request handler that uses it (example: [2]). This keeps the DB transaction from committing until _after_ the handler writes the API response. Since a DB transaction uses limited shared resources, it should be closed as soon as possible after the required DB operation in order to return the used DB connection back to the pool for other request handlers to use.

Performance testing may be needed in order to prioritize this based upon performance impact. A good place to start would be with shared_handlers.go which would affect all the CRUD handlers.

[1] https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/api/api.go#L268
[2] https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/api/shared_handlers.go#L130

**Comment by rob05c** [comment]
Yeah, I've been considering changing `*inf.CommitTx = true` to a function,` inf.Commit()`, which actually commits, and then the deferred `Close` will do nothing if it's already been committed.

That will let it be committed as soon as possible, which is good for integrity, committing before it's returned to the user. I'm skeptical whether it's a performance issue, but it'll address that as well.
